superset/utils/cache_manager.py

```diff
@@ -15,15 +15,40 @@
 # specific language governing permissions and limitations
 # under the License.
 import logging
+from typing import Any, Optional, Union
 
 from flask import Flask
 from flask_caching import Cache
+from markupsafe import Markup
+
+from superset.utils.core import DatasourceType
 
 logger = logging.getLogger(__name__)
 
 CACHE_IMPORT_PATH = "superset.extensions.metastore_cache.SupersetMetastoreCache"
 
 
+class ExploreFormDataCache(Cache):
+    def get(self, *args: Any, **kwargs: Any) -> Optional[Union[str, Markup]]:
+        cache = self.cache.get(*args, **kwargs)
```

Comment:
There's a problem here with old keys because the `contextual_key` is now being generated with the datasource_type but the old keys didn't have it so it will miss the cache and return `None` for old keys.

Reply:
Ok, I added a check using the old keys for update and delete as well. I don't really think there's a good way to remap these, unfortunately. The cache persistence is configurable but the default is 7 days.. maybe we can delete this fallback code after three months or so.